src/classes/Contacts.cls

```diff
@@ -0,0 +1,713 @@
+    }
+
+    private static npo02__Household__c getHouseholdForInsertFromContact(Contact c) {
+        npo02__household__c h = new npo02__household__c();
+        String hName = c.LastName;
+        hName += ' ' + system.label.npo02.DefaultHouseholdName;
+        h.Name = hName;
+        //sync the primary address block
+        h.npo02__MailingStreet__c = c.MailingStreet;
+        h.npo02__MailingCity__c = c.MailingCity;
+        h.npo02__MailingState__c = c.MailingState;
+        h.npo02__MailingPostalCode__c = c.MailingPostalCode;
+        h.npo02__MailingCountry__c = c.MailingCountry;
+        h.npo02__HouseholdPhone__c = c.HomePhone;
+        h.npo02__HouseholdEmail__c = c.Email;
+        return h;
+    }
+
+    /*******************************************************************************************************
+    * @description Determines if a Contact Should have a household object created
+    * @param contacts List of Contacts
+    * @param isAfterTrigger Whether called from the afterInsert trigger (vs. afterUpdate trigger) on Contacts
+    * @param dmlWrapper Not used. Commits are immediate.
+    * @return void
+    ********************************************************************************************************/
+    public void insertHouseholdsAfterInsert(Contact[] contacts) {
+        insertHouseholdsForContacts(contacts, true);
+    }
+
+    private Map<Id, Account> getContactAccountMap(Contact[] contacts) {
+        List<Id> accountIds = getAccountIds(contacts);
+        Map<Id, Account> contactAccountMap = new Map<Id, Account>([Select Id, npe01__SYSTEMIsIndividual__c, Name from Account where Id in :accountIds]);
+        return contactAccountMap;
+    }
+
+    private List<Id> getAccountIds(Contact[] contacts) {
+        List<Id> accountIds = new List<Id>();
+        for (Contact c : contacts) {
+            if (c.accountId != null) {
+                accountIds.add(c.AccountId);
+            }
+        }
+        return accountIds;
+    }
+
+    private List<Contact> getContactsNeedingHouseholds(Contact[] contacts) {
+        //get all the Accounts so we can test to see if they are individuals
+        Map<Id, Account> contactAccountMap = getContactAccountMap(contacts);
+
+        List<Contact> householdInserts = new List<Contact>();
+        for (Contact c : contacts) {
+            if (c.AccountId != null) {
+                Account acc = contactAccountMap.get(c.AccountId);
+                //check the system field on Account to see if the account is an individual
+                if (acc.npe01__SYSTEMIsIndividual__c == true) {
+                    householdInserts.add(c);
+                }
+            } else if (c.AccountId == null) {
+                householdInserts.add(c);
+            }
+        }
+        return householdInserts;
+    }
+
+    List<npo02__Household__c> emptyHouseholdsToDelete = new List<npo02__Household__c>();
```

Comment:
Class properties should be defined at the top of the class and specified as private.